src/installer/corehost/corehost.cpp

```diff
@@ -218,6 +218,7 @@ int exe_start(const int argc, const pal::char_t* argv[])
 
             rc = main_fn_v2(argc, argv, host_path_cstr, dotnet_root_cstr, app_path_cstr);
 
+#if defined(_WIN32) && defined(FEATURE_APPHOST)
```

Comment:
While the ifdef is correct it does not guarantee that we will only do this for GUI apps. Win32 console app will still try to do this as well. I think the right check is to only do this if the apphost error writing is redirected (so check that the g_error_writer in trace.cpp is not null.